src/catalog/database_catalog.cpp

```diff
@@ -785,6 +789,7 @@ table_oid_t DatabaseCatalog::GetTableOid(transaction::TransactionContext *const
 
 bool DatabaseCatalog::SetTablePointer(transaction::TransactionContext *const txn, const table_oid_t table,
                                       const storage::SqlTable *const table_ptr) {
+  if (!TryLock(txn)) return false;
```

Comment:
This should probably be replaced by an assertion that the current transaction already owns the lock since this should only occur after a call to `CreateTable`.

Reply:
Breaks Bootstrap because it calls directly to the internal CreateTableEntry API. Bah.